overlord/devicestate/fde/fde.go

```diff
@@ -0,0 +1,79 @@
+// -*- Mode: Go; indent-tabs-mode: t -*-
+
+/*
+ * Copyright (C) 2020  Canonical Ltd
+ *
+ * This program is free software: you can redistribute it and/or modify
+ * it under the terms of the GNU General Public License version 3 as
+ * published by the Free Software Foundation.
+ *
+ * This program is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ * GNU General Public License for more details.
+ *
+ * You should have received a copy of the GNU General Public License
+ * along with this program.  If not, see <http://www.gnu.org/licenses/>.
+ *
+ */
+
+// package fde implements helper used by low level parts like secboot
+// in snap-bootstrap and high level parts like DeviceManager in snapd.
+//
+// Note that it must never import anything overlord related itself
+// to avoid increasing the size of snap-bootstrap.
+package fde
+
+import (
+	"os/exec"
+
+	"github.com/snapcore/snapd/secboot"
+)
+
+func init() {
+	secboot.FDEHasRevealKey = HasRevealKey
+}
+
+// HasRevealKey return true if the current system has a "fde-reveal-key"
+// binary (usually used in the initrd).
+//
+// This will be setup by devicestate to support device-specific full
+// disk encryption implementations.
+func HasRevealKey() bool {
+	// XXX: should we record during initial sealing that the fde-setup
+	//      was used and only use fde-reveal-key in that case?
+	_, err := exec.LookPath("fde-reveal-key")
+	return err == nil
```

Comment:
Given how hard this area has been to debug if something went south, would it make sense to report/log the error? LookPath may return ErrNotFound, os.ErrPermission or something else. I think it may be useful to know what it was exactly in case the hook exists, but has no exec permissions.

Reply:
There is also a case above (I was thinking about hitting that one potentially): 
```
if strings.Contains(file, "/") {
		err := findExecutable(file)
		if err == nil {
			return file, nil
		}
		return "", &Error{file, err}. <-- may return os.ErrPermission
	}
```

Reply:
But maybe it's not relevant if we expect to be looking in PATH, which is the code below as you said... Feel free to ignore if that's the case.

Reply:
Yeah, I did see that - but it only applies for file having "/" - we only look for "fde-reveal-key" so it seems like we won't hit it (unless we look at a hardcodd path).

Reply:
Yeah, don't get me wrong, the suggestion is great. Ff we would get os.ErrPermission for matching files in PATH that have the wrong permissions I think this is totally  worth it. But if all we get is ErrNotExists it's not great. I think I would rather implement our own LookPath() then to provide useful log info.